спринт 12 домашка 1

AleksandrRegulus opened the pull request.

AleksandrRegulus committed:
спринт 12 домашка 1

AleksandrRegulus committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	app/src/main/res/layout/track_item.xml

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -70,6 +119,9 @@ class SearchActivity : AppCompatActivity() {
             override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                 searchText = binding.searchEditText.text.toString()
                 binding.btnClear.visibility = clearButtonVisibility(s)
+//                binding.searchEditText.hasFocus() &&
```
🍏 Неиспользуемый код лучше удалять перед пулл реквестом

AleksandrRegulus replied:
учту, спасибо

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -26,14 +26,55 @@ class SearchActivity : AppCompatActivity() {
+        val onTrackClickListener = object : TracksAdapter.OnItemClickListener {
+            override fun onItemClick(track: Track) {
+                val readTracks = searchHistory.readHistory()
+                val position = readTracks.indexOf(track)
```
🍏 Рекомендую спрятать всю логику по добавлению треков в избранное в SearchHistory

AleksandrRegulus replied:
исправил

kgornostaeva commented:
Отображение продолжительности трека должно располагаться сразу после точки, а в приложении оно уезжает вправо к стрелочке перехода.

<img width="462" alt="Снимок экрана 2023-08-12 в 18 42 12" src="https://github.com/AleksandrRegulus/PlaylistMaker/assets/48408134/2f3e2627-d298-4c4f-918a-c970d24c3abb">

kgornostaeva commented:
На экране настроек иконка стрелочки напротив "Пользовательского соглашения" не соответствует макетам, более вытянутая. Она должна быть такой же, как и в списке на экране поиска.

<img width="462" alt="Снимок экрана 2023-08-12 в 18 40 40" src="https://github.com/AleksandrRegulus/PlaylistMaker/assets/48408134/188323db-afda-4489-89eb-9f91d054f157">

AleksandrRegulus committed:
спринт 12 домашка 2

1. Переделал макет элементов списка треков на constraintlayout. Теперь длина трека расположена корректно
2.Исправил стрелку в настройках
3.Вынес логику добавления треков в класс SearchHistory

kgornostaeva commented on .idea/gradle.xml:
```diff
@@ -7,6 +7,7 @@
         <option name="testRunner" value="GRADLE" />
```
🍏 Добавь всю папку /.idea в файл  .gitignore, чтобы она не подгружалась в Pull request

AleksandrRegulus replied:
сделал

kgornostaeva commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -44,24 +46,18 @@ class SearchActivity : AppCompatActivity() {
         val onHistoryTrackClickListener = object : TracksAdapter.OnItemClickListener {
-            override fun onItemClick(track: Track) {}
+            override fun onItemClick(track: Track) {
+                searchHistory.addTrackToHistoy(track)
```
🍏 Закралась опечатка:
```suggestion
                searchHistory.addTrackToHistory(track)
```

AleksandrRegulus replied:
пофиксил, спасибо)

AleksandrRegulus committed:
спринт 12 домашка 3

AleksandrRegulus committed:
спринт 12 домашка 4

AleksandrRegulus committed:
спринт 12 домашка 5
